### ISSUE TITLE
Error in data stream

### ISSUE THREAD

**Issue by Leo4815162342** [Leo4815162342/dukascopy-tools#7]
```
const config = { instrument: 'audcad',   dates: { from: '2010-01-01 00:00', to: '2020-04-13 22:26' },   timeframe: 'h1'}
```

**Comment by lezwright** [comment]
Would it help if we have an account with Dukascopy? Because we have an account with them and we get our live data from them.

**Comment by lezwright** [comment]
Hi Leo,

Thanks for your email. Please see the answer to your questions below:

"Can you tell me how you use the dukascopy-node package?"
We use it to feed the data into our customised Tradingview charts. As I
mentioned in my previous email we have the live data account with
Dukascopy, that is we can get live tick data (FIX API). They also told us
it is possible to get the historical data from them through JFOREX API
https://www.dukascopy.com/wiki/en/development/strategy-api/historical-data
That is why I inquired about having an account and log in id might remove
the limitation on downloading the data.

"Do you often search for historical data for the same instrument? How many
calls you make?"
I need to have access to all 61 FX currency pairs (including gold and
silver) from the Forex list you provided on the library home page:
https://github.com/Leo4815162342/dukascopy-tools for the entire historical
data that is available.
I call the data in R for the entire historical data that is available for
each currency pair. On one currency pair I might need to make several calls
during a day (150+ requests each time).

Is it possible to remove any limitation on the number of requests made?

Many thanks

**Comment by lezwright** [comment]
My currency pairs have stopped working one by one, starting GBPJPY then AUDCAD?

**Comment by Leo4815162342** [comment]
Can you specify exactly what is not working?
I've just made few test call, and it looks fine

**Comment by lezwright** [comment]
I couldn't get the data for GBPJPY and AUDCAD but it is fine now, Leo.

**Comment by Leo4815162342** [comment]
@lezwright 
The `dukascopy-node` package does not have any limitations on the number of calls it can make.

The only limitation is the dukascopy server rate limits - which we cannot control. But we can control how we make our calls.

I've released an update (`dukascopy-nod@1.2.0`: https://www.npmjs.com/package/dukascopy-node) that exposes more controls in terms of how we call dukascopy servers:

 - `batchsize`: https://github.com/Leo4815162342/dukascopy-tools/tree/master/packages/dukascopy-node#batchsize

 - `pauseBetweenBatchesMs`: https://github.com/Leo4815162342/dukascopy-tools/tree/master/packages/dukascopy-node#pausebetweenbatchesms

You can play around with those, for example decrease `batchsize` and increase `pauseBetweenBatchesMs` - with that you can have higher chances of all requests coming through

For example:
```javascript
getHistoricRates({
  instrument: 'audcad',
  dates: {
    from: '2010-01-01',
    to: '2020-01-01'
  },
  timeframe: 'h1',
  batchSize: 5,
  pauseBetweenBatchesMs: 2500
})
```

**Comment by lezwright** [comment]
Thank you very much Leo! I will give it a go and update you.
